src/functionaltests/java/com/ericsson/ei/notifications/ttl/TestTTLSteps.java

```diff
@@ -0,0 +1,225 @@
+            mockServerClient.close();
+            clientAndServer.stop();
+        }
+
+        // START TEST SCENARIOS
+
+        @Given("^Missed notification is created in database with index \"([A-Za-z0-9_]+)\"$")
+        public void missed_notification_is_created_in_database(String indexName) throws IOException, ParseException {
+            LOGGER.debug("Starting scenario @TestTTL");
+            JsonNode missedNotification = getJSONFromFile(MISSED_NOTIFICATION_FILE_PATH);
+            BasicDBObject missedNotificationDocument = prepareDocumentWithIndex(missedNotification, indexName);
+
+            // setting 1 second TTL on index in db
+            mongoDBHandler.createTTLIndex(missedNotificationDatabase, missedNotificationCollectionName,
+                    indexName, 1);
+            Boolean isInserted = mongoDBHandler
+                    .insertDocument(missedNotificationDatabase,
+                            missedNotificationCollectionName,
+                            missedNotificationDocument.toString());
+            assertEquals("Failed to create missed notification in database", true, isInserted);
+
+            List<String> result = mongoDBHandler.getAllDocuments(missedNotificationDatabase,
+                    missedNotificationCollectionName);
+            assertEquals("Database " + missedNotificationDatabase + " is not empty.", false, result.isEmpty());
+        }
+
+        @Given("^Aggregated object is created in database with index \"([A-Za-z0-9_]+)\"$")
+        public void aggregated_object_is_created_in_database(String indexName) throws IOException, ParseException {
+
+            JsonNode aggregatedObject = getJSONFromFile(AGGREGATED_OBJECT_FILE_PATH);
+            BasicDBObject aggregatedDocument = prepareDocumentWithIndex(aggregatedObject, indexName);
+
+            // setting 1 second TTL on index in db
+            mongoDBHandler.createTTLIndex(aggregationDataBaseName, aggregationCollectionName,
+                    indexName, 1);
+            Boolean isInserted = mongoDBHandler.insertDocument(aggregationDataBaseName,
+                    aggregationCollectionName, aggregatedDocument.toString());
+            assertEquals("Failed to create aggregated object in database",true, isInserted);
+
+            List<String> result = mongoDBHandler.getAllDocuments(aggregationDataBaseName,
+                    aggregationCollectionName);
+            assertEquals("Database " + aggregationDataBaseName + " is not empty.", false, result.isEmpty());
+        }
+
+        @When("^I sleep for \"([^\"]*)\" seconds")
+        public void i_sleep_for_time(int sleepTime) throws Throwable {
+            LOGGER.debug("Sleeping for " + sleepTime + " at " + LocalTime.now());
+            // The background task deleting notifications from Mongo DB
+            // runs every 60 seconds
+            TimeUnit.SECONDS.sleep(sleepTime);
+        }
+
+        @Then("^\"([^\"]*)\" has been deleted from \"([A-Za-z0-9_]+)\" database$")
```

Comment:
Really hard for understand what is going to be deleted in test step descirption nor the code. You can adde something general description before regex, e.g. SubscriptionName "^\"([^\"]*)\"", then it would be more clearer what is happening here.

Reply:
Hmm, this test step is general because a check is done to verify both aggregated object and missed notification has been deleted, that's why I couldn't find a general noun to fit both in the  method name. 
An option is to directly spell out missed notification and aggregated object into the test steps and check both databases at the same time and not have regex in feature.